bottlecap/src/bin/bottlecap/main.rs

```diff
@@ -375,92 +383,95 @@ async fn extension_loop_active(
                         Event::Metric(event) => {
                             debug!("Metric event: {:?}", event);
                         }
-                        Event::Telemetry(event) => match event.record {
-                            TelemetryRecord::PlatformStart { request_id, .. } => {
-                                invocation_context_buffer.insert(InvocationContext {
-                                    request_id,
-                                    runtime_duration_ms: 0.0,
-                                });
-                            }
-                            TelemetryRecord::PlatformInitReport {
-                                initialization_type,
-                                phase,
-                                metrics,
-                            } => {
-                                debug!("Platform init report for initialization_type: {:?} with phase: {:?} and metrics: {:?}", initialization_type, phase, metrics);
-                                lambda_enhanced_metrics
-                                    .set_init_duration_metric(metrics.duration_ms);
-                            }
-                            TelemetryRecord::PlatformRuntimeDone {
-                                request_id,
-                                status,
-                                metrics,
-                                ..
-                            } => {
-                                if let Some(metrics) = metrics {
-                                    invocation_context_buffer
-                                        .add_runtime_duration(&request_id, metrics.duration_ms);
+                        Event::Telemetry(event) =>
+                            match event.record {
+                                TelemetryRecord::PlatformStart { request_id, .. } => {
+                                    let mut p = invocation_processor.lock().await;
```

Comment:
do we need a lock here? this prevents us from processing any messages until it's acquired, what's it fully protecting here?

Reply:
Looking what improvements I can make here

Reply:
As per our conversation, will move the lock out of this in another PR when the time comes, will add a ticket for this